lib/processEventDataHelper.js

```diff
@@ -1,105 +1,100 @@
-
-    if (cfg.importance) {
-        result.importance = cfg.importance;
-    }
-
-    if (cfg.sensitivity) {
-        result.sensitivity = cfg.sensitivity;
-    }
-
-    if (cfg.showAs) {
-        result.showAs = cfg.showAs;
-    }
-
-    if (cfg.bodyContentType) {
-        result.body.contentType = cfg.bodyContentType;
-    }
-
-    result.end.timeZone = cfg.timeZone;
-    result.start.timeZone = cfg.timeZone;
-
-    if ((cfg.isAllDay) && (cfg.isAllDay === 'true')) {
-        result.isAllDay = cfg.isAllDay;
-        // The last day given as parameter is exclusive.
-        // If the user wants to add all day event for 21-23 for ex,
-        // outlook expects something like: 21 (00:00 AM) - 24 (00:00 AM)
-        let FORMAT = 'YYYY-MM-DD';
-        result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
-        result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
-        result.end.dateTime = moment(result.end.dateTime).add(1,'days').format(FORMAT);
-    } else {
-        let FORMAT = 'YYYY-MM-DDTHH:mm:ss';
-        result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
-        result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
-    }
-
-    return result;
+  checkRequiredFields(cfg, messageBody);
+
+  const result = _.cloneDeep(messageBody);
+
+  if (cfg.importance) {
+    result.importance = cfg.importance;
+  }
+
+  if (cfg.sensitivity) {
+    result.sensitivity = cfg.sensitivity;
+  }
+
+  if (cfg.showAs) {
+    result.showAs = cfg.showAs;
+  }
+
+  if (cfg.bodyContentType) {
+    result.body.contentType = cfg.bodyContentType;
+  }
+
+  result.end.timeZone = cfg.timeZone;
+  result.start.timeZone = cfg.timeZone;
+
+  if ((cfg.isAllDay) && (cfg.isAllDay === 'true')) {
+    result.isAllDay = cfg.isAllDay;
+    // The last day given as parameter is exclusive.
+    // If the user wants to add all day event for 21-23 for ex,
+    // outlook expects something like: 21 (00:00 AM) - 24 (00:00 AM)
+    const FORMAT = 'YYYY-MM-DD';
+    result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
+    result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
+    result.end.dateTime = moment(result.end.dateTime).add(1, 'days').format(FORMAT);
+  } else {
+    const FORMAT = 'YYYY-MM-DDTHH:mm:ss';
+    result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
+    result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
+  }
+
+  return result;
 }
 
 function processEventData(cfg, messageBody) {
-    console.log('Processing event data ...');
-    var defer = Q.defer();
-    try {
-        let result = buildPostBody(cfg, messageBody);
-        defer.resolve(result);
-    } catch (err) {
-        defer.reject(err);
-    }
-    return defer.promise;
+  console.log('Processing event data ...');
```

Comment:
Should we replace this `console.log` with `this.logger`?

Reply:
done